model/index.js

```diff
@@ -165,13 +166,13 @@ module.exports = class Model extends EventEmitter {
   handleResponse(response, callback) {
     let data = {};
     try {
-      data = JSON.parse(response.body || '{}');
+      data = typeof response.data === 'object' ? response.data : JSON.parse(response.data || '{}');
     } catch (err) {
-      err.status = response.statusCode;
-      err.body = response.body;
-      return callback(err, null, response.statusCode);
+      err.status = response.status;
+      err.body = response.data;
+      return callback(err, null, response.status);
     }
-    return this.parseResponse(response.statusCode, data, callback);
+    return this.parseResponse(response.status, data, callback);
   }
 
```

Comment:
it would be nice to have the code written in async/await format.  Not sure if this will work

```suggestion
async handleResponse(response) {
  let data = null;

  try {
    // Attempt to parse the response data
    if (typeof response.data === 'object') {
      data = response.data;
    } else if (typeof response.data === 'string' && response.data.trim() !== '') {
      data = JSON.parse(response.data);
    } else {
      data = {};
    }
  } catch (err) {
    err.message = 'Failed to parse response data';
    err.status = response.status;
    err.body = response.data;
    throw err;
  }

  return this.parseResponse(response.status, data);
}
```

Reply:
I have converted to async/await and updated the tests as well. Have tested in some of the services and it is working, but see what you think

Reply:
Wow, thank you. I was a bit concerned about mentioning it to you as I didn't want to overload you